devtools/common/src/main/java/io/quarkus/cli/commands/AddExtensions.java

```diff
@@ -24,44 +26,64 @@ public AddExtensions(final File pom) throws IOException {
-            Optional<Extension> optional = MojoUtils.loadExtensions().stream()
-                    .filter(d -> {
-                        boolean hasTag = d.labels().contains(dependency.trim().toLowerCase());
-                        boolean machName = d.getName()
-                                .toLowerCase()
-                                .contains(dependency.trim().toLowerCase());
-                        return hasTag || machName;
-                    })
-                    .findAny();
-
-            if (optional.isPresent()) {
-                final Extension extension = optional.get();
-
-                if (!MojoUtils.hasDependency(model, extension.getGroupId(), extension.getArtifactId())) {
-                    System.out.println("Adding extension " + extension.managementKey());
-                    model.addDependency(extension
-                            .toDependency(containsBOM(model) &&
-                                    isDefinedInBom(dependenciesFromBom, extension)));
-                    updated = true;
+            Dependency res;
+            if (!dependency.contains(":")) {
+                String groupId, artifactId;
+                groupId = QUARKUS_GROUPID;
+                if (!dependency.startsWith(EXTENSION_ARTIFACTID_PREFIX)) {
+                    artifactId = EXTENSION_ARTIFACTID_PREFIX + dependency.trim().toLowerCase();
                 } else {
-                    System.out.println("Skipping extension " + extension.managementKey() + ": already present");
+                    artifactId = dependency.trim().toLowerCase();
                 }
-            } else if (dependency.contains(":")) {
-                Dependency parsed = MojoUtils.parse(dependency);
-                System.out.println("Adding dependency " + parsed.getManagementKey());
-                model.addDependency(parsed);
+                res = new Dependency();
+                res.setGroupId(groupId);
+                res.setArtifactId(artifactId);
+            } else {
+                if (dependency.startsWith(QUARKUS_GROUPID)) {
+                    dependency = dependency.toLowerCase();
+                }
+                res = MojoUtils.parse(dependency.trim());
+            }
+
+            if (QUARKUS_GROUPID.equals(res.getGroupId())) {
+                Optional<Extension> optional = loadedExtensions.stream()
+                        .filter(d -> {
+                            return d.getArtifactId()
+                                    .equals(res.getArtifactId());
+                        })
+                        .findAny();
+
+                if (!optional.isPresent()) {
+                    throw new CommandsException("Cannot find a Quarkus extension matching " + dependency);
+                } else if ((res.getVersion() == null || res.getVersion().isEmpty()) && !containsBOM(model)
+                        && isDefinedInBom(dependenciesFromBom, res)) {
+                    res.setVersion(optional.get().getVersion());
```

Comment:
Wouldn't it make sense to add the version all the time here (if the version is not set)...

Reply:
No because, if it's not set but we have a dependencyManagement we want to use it and in the pom the version must not be defined. This way the user can upgrade quarkus lib by just upgrade the quarkus-bom

Reply:
remarks, the extension version set from extension.json is ${quarkus.version} so i can set it in all case. As you want

Reply:
Yeah, I would set it here in all cases and have the logic about the bom only once.

Reply:
is it ok now ?